nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -453,6 +453,17 @@ sub in {
         }
     }
 
+    # Preserve umask (fmask, dmask) settings for vfat filesystems.
+    # (The default is to mount these world-readable, but that's a security risk
+    # for the EFI System Partition.)
+    if ($fsType eq "vfat") {
+        for (@superOptions) {
+            if ($_ =~ /fmask|dmask/) {
```

Comment:
```suggestion
            if ($_ =~ /fmask|dmask|umask/) {
```

Reply:
Oh I guess that can't occur there, never mind.

Reply:
Yes, it seems so. (I started looking for `umask`, then noticed that `/proc/self/mountinfo` represents `umask` as  `fmask` + `dmask`.)